### ISSUE TITLE
Quick find / Ctrl+F3 highlight regression (in Javascript in HTML files)

### ISSUE THREAD

**Issue by elonmallin** [Microsoft/vscode#13493]
**EDIT: JS inside HTML file**

If you place the cursor on a string `"hello"` and press `ctrl+f3` the quick find will mark `"hello"` including quotes. This is different from previous versions. Worse is if you do the same for a function definition in an object `_myFuncName: function() {}` the quick find will be `_myFuncName:` which includes the `:`.

Last example:
`this.set('something.bla.' + i + '.prop', 1);` place the cursor on 'something' and pressing `ctrl+f3` will highlight `this.set('something.bla.'`.

- VSCode Version:
1.6.0
- OS Version:
Windows 10

**Comment by jrieken** [comment]
Did we change the word definition?

**Comment by dbaeumer** [comment]
No I didn't. The definition is as follows right now:

```ts
			languages.setLanguageConfiguration(modeId, {
				indentationRules: {
					// ^(.*\*/)?\s*\}.*$
					decreaseIndentPattern: /^(.*\*\/)?\s*\}.*$/,
					// ^.*\{[^}"']*$
					increaseIndentPattern: /^.*\{[^}"']*$/
				},
				wordPattern: /(-?\d*\.\d\w*)|([^\`\~\!\@\#\%\^\&\*\(\)\-\=\+\[\{\]\}\\\|\;\:\'\"\,\.\<\>\/\?\s]+)/g,
				onEnterRules: [
					{
						// e.g. /** | */
						beforeText: /^\s*\/\*\*(?!\/)([^\*]|\*(?!\/))*$/,
						afterText: /^\s*\*\/$/,
						action: { indentAction: IndentAction.IndentOutdent, appendText: ' * ' }
					},
					{
						// e.g. /** ...|
						beforeText: /^\s*\/\*\*(?!\/)([^\*]|\*(?!\/))*$/,
						action: { indentAction: IndentAction.None, appendText: ' * ' }
					},
					{
						// e.g.  * ...|
						beforeText: /^(\t|(\ \ ))*\ \*(\ ([^\*]|\*(?!\/))*)?$/,
						action: { indentAction: IndentAction.None, appendText: '* ' }
					},
					{
						// e.g.  */|
						beforeText: /^(\t|(\ \ ))*\ \*\/\s*$/,
						action: { indentAction: IndentAction.None, removeText: 1 }
					},
					{
						// e.g.  *-----*/|
						beforeText: /^(\t|(\ \ ))*\ \*[^/]*\*\/\s*$/,
						action: { indentAction: IndentAction.None, removeText: 1 }
					}
				]
			});

```

Looking at the regexp it correctly excludes " and '. May be another extensions ?

**Comment by elonmallin** [comment]
Oh wait I'm in an HTML file within a script tag actually, sorry. So in a `.js` file it seems to work.
I know there is some issue about using js and css in html files (something like languages in other languages or something) but this behavior didn't happen in the previous version so still seems like something changed.

Is there some way to start VSCode without loading any extensions. I could make sure it's not any extension doing this also.

**Comment by jrieken** [comment]
@aeschli I know you use a crazy word definition for html

**Comment by aeschli** [comment]
- we need to bring back the embedded modes so that the JS word regex is used again in embedded JS (#11081)
- we need to untangle the uses of the word regex. It's used for highlighting, selection and for code assist default replace ranges. The last one should have a separate regex. Or better, we should move away from the 'currentWord'.
- I still think Ctrl+F3 should use the same word that you get when double clicking (based on word separators)
- I'll look into the html word regex. It is actually really too crazy, I agree.

**Comment by aeschli** [comment]
Fixed by #13675
